### ISSUE TITLE
AspNet Instrumentation OnException hook

### ISSUE THREAD

**Issue by christopher-taormina-zocdoc** [open-telemetry/opentelemetry-dotnet#1701]
# Question

Is there a reason that AspNet instrumentation does not have an OnException hook? I've noticed that unlike the AspNetCore instrumentation, aspnet does not set the activity status when an exception occurs.

Describe any aspect of your environment relevant to the question.
AspNet Instrumentation

**What are you trying to achieve?**
Update the activity when an exception occurs.

What did you expect to see?
The activity status updated on exception

## Additional Context

Add any other context about the problem here. If you followed an existing
documentation, please share the link to it.

**Comment by cijothomas** [comment]
https://github.com/aspnet/Microsoft.AspNet.TelemetryCorrelation/blob/master/src/Microsoft.AspNet.TelemetryCorrelation/TelemetryCorrelationHttpModule.cs Doesn't look like this is firing an OnException event. We can explore if we have a way to obtain exception.

Btw we still set status based on HttpResponse code, right?

**Comment by christopher-taormina-zocdoc** [comment]
@cijothomas yeah, setstatus is still called with the response code if the status is unset https://github.com/open-telemetry/opentelemetry-dotnet/blob/a16c14ce6a579f10109567dd14301af3cf9f7382/src/OpenTelemetry.Instrumentation.AspNet/Implementation/HttpInListener.cs#L181

**Comment by cijothomas** [comment]
Adding support for this in asp.net correlation module https://github.com/aspnet/Microsoft.AspNet.TelemetryCorrelation/pull/88
